astropy v7.0 incompatibility in TestMasksTracing.test_fit_trace_all_nan_cols

According to [astropy-integration-testing run](https://github.com/astropy/astropy-integration-testing/actions/runs/11785052451/job/32825493170), there is failure with both stable and dev versions, so this package needs a compatibility fix and a new release to be compatible with `astropy` v7.0.0. FYI.

```
>       np.testing.assert_allclose(truth, max_trace.trace)
E       AssertionError: 
E       Not equal to tolerance rtol=1e-07, atol=0
E       
E       Mismatched elements: 11 / 11 (100%)
E       Max absolute difference among violations: 0.00700257
E       Max relative difference among violations: 0.00360873
E        ACTUAL: array([1.947455, 2.383634, 2.819813, 3.255992, 3.692171, 4.12835 ,
E              4.564529, 5.000708, 5.436887, 5.873067, 6.309246])
E        DESIRED: MaskedArray([1.940452, 2.376862, 2.813273, 3.249683, 3.686093, 4.122503,
E                    4.558913, 4.995323, 5.431733, 5.868143, 6.304553])
```

This test already ignores a bunch of warnings, so maybe not a well designed test to begin with. Just increase the tolerance?

Also appears to be using a fitter that is no longer recommended, see:

* https://github.com/astropy/astropy/pull/16983

--- a/CHANGES.rst
+++ b/CHANGES.rst
@@ -10,24 +10,32 @@ API Changes
 Bug Fixes
 ^^^^^^^^^
 
+Other changes
+^^^^^^^^^^^^^
+
+1.4.2
+------
+
+Bug Fixes
+^^^^^^^^^
+- Fixed Astropy v7.0 incompatibility bug [#229] in ``tracing.FitTrace``: changed to use
+  ``astropy.modeling.fitting.DogBoxLSQFitter`` when fitting a Gaussian peak model instead of
+  ``astropy.modeling.fitting.LevMarLSQFitter`` that may be deprecated in the future. Also
+  changed to use ``fitting.LMLSQFitter`` instead of ``fitting.LevMarLSQFitter`` when fitting
+  a generic nonlinear trace model.
 
 Other changes
+^^^^^^^^^^^^^
+- Changed ``tracing.FitTrace`` to use ``astropy.modeling.fitting.LinearLSQFitter``
+  if the trace model is linear.
 
 1.4.1 (2024-06-20)
 ------------------
-
-New Features
-^^^^^^^^^^^^
-
-API Changes
-^^^^^^^^^^^
 
 Bug Fixes
 ^^^^^^^^^
 - Fix bug where Background one sided / two sided was not correctly assigning units to data. [#221]
 
-Other changes
-^^^^^^^^^^^^^
 
 1.4.0 (2024-05-29)
 ------------------

--- a/specreduce/tests/test_tracing.py
+++ b/specreduce/tests/test_tracing.py
@@ -203,21 +203,21 @@ class TestMasksTracing():
                  4.6474359, 5.25, 5.8525641, 6.4551282, 7.0576923,
                  7.6602564]
         max_trace = FitTrace(img, peak_method='max')
-        np.testing.assert_allclose(truth, max_trace.trace, atol=0.05, rtol=1e-2)
+        np.testing.assert_allclose(truth, max_trace.trace, atol=0.05)
 
         # peak_method = 'gaussian'
         truth = [1.947455, 2.383634, 2.8198131, 3.2559921, 3.6921712,
                  4.1283502, 4.5645293, 5.0007083, 5.4368874, 5.8730665,
                  6.3092455]
         max_trace = FitTrace(img, peak_method='gaussian')
-        np.testing.assert_allclose(truth, max_trace.trace, atol=0.05, rtol=1e-2)
+        np.testing.assert_allclose(truth, max_trace.trace, atol=0.05)
 
         # peak_method = 'centroid'
         truth = [2.5318835, 2.782069, 3.0322546, 3.2824402, 3.5326257,
                  3.7828113, 4.0329969, 4.2831824, 4.533368, 4.7835536,
                  5.0337391]
         max_trace = FitTrace(img, peak_method='centroid')
-        np.testing.assert_allclose(truth, max_trace.trace, atol=0.05, rtol=1e-2)
+        np.testing.assert_allclose(truth, max_trace.trace, atol=0.05)
 
     @pytest.mark.filterwarnings("ignore:The fit may be unsuccessful")
     @pytest.mark.filterwarnings("ignore:Model is linear in parameters")

--- a/specreduce/tests/test_tracing.py
+++ b/specreduce/tests/test_tracing.py
@@ -203,21 +203,21 @@ class TestMasksTracing():
                  4.6474359, 5.25, 5.8525641, 6.4551282, 7.0576923,
                  7.6602564]
         max_trace = FitTrace(img, peak_method='max')
-        np.testing.assert_allclose(truth, max_trace.trace)
+        np.testing.assert_allclose(truth, max_trace.trace, atol=0.05, rtol=1e-2)
 
         # peak_method = 'gaussian'
         truth = [1.947455, 2.383634, 2.8198131, 3.2559921, 3.6921712,
                  4.1283502, 4.5645293, 5.0007083, 5.4368874, 5.8730665,
                  6.3092455]
         max_trace = FitTrace(img, peak_method='gaussian')
-        np.testing.assert_allclose(truth, max_trace.trace)
+        np.testing.assert_allclose(truth, max_trace.trace, atol=0.05, rtol=1e-2)
 
         # peak_method = 'centroid'
         truth = [2.5318835, 2.782069, 3.0322546, 3.2824402, 3.5326257,
                  3.7828113, 4.0329969, 4.2831824, 4.533368, 4.7835536,
                  5.0337391]
         max_trace = FitTrace(img, peak_method='centroid')
-        np.testing.assert_allclose(truth, max_trace.trace)
+        np.testing.assert_allclose(truth, max_trace.trace, atol=0.05, rtol=1e-2)
 
     @pytest.mark.filterwarnings("ignore:The fit may be unsuccessful")
     @pytest.mark.filterwarnings("ignore:Model is linear in parameters")

--- a/specreduce/tests/test_tracing.py
+++ b/specreduce/tests/test_tracing.py
@@ -203,21 +203,21 @@ class TestMasksTracing():
                  4.6474359, 5.25, 5.8525641, 6.4551282, 7.0576923,
                  7.6602564]
         max_trace = FitTrace(img, peak_method='max')
-        np.testing.assert_allclose(truth, max_trace.trace, atol=0.05)
+        np.testing.assert_allclose(truth, max_trace.trace, atol=0.1)
 
         # peak_method = 'gaussian'
         truth = [1.947455, 2.383634, 2.8198131, 3.2559921, 3.6921712,
                  4.1283502, 4.5645293, 5.0007083, 5.4368874, 5.8730665,
                  6.3092455]
         max_trace = FitTrace(img, peak_method='gaussian')
-        np.testing.assert_allclose(truth, max_trace.trace, atol=0.05)
+        np.testing.assert_allclose(truth, max_trace.trace, atol=0.1)
 
         # peak_method = 'centroid'
         truth = [2.5318835, 2.782069, 3.0322546, 3.2824402, 3.5326257,
                  3.7828113, 4.0329969, 4.2831824, 4.533368, 4.7835536,
                  5.0337391]
         max_trace = FitTrace(img, peak_method='centroid')
-        np.testing.assert_allclose(truth, max_trace.trace, atol=0.05)
+        np.testing.assert_allclose(truth, max_trace.trace, atol=0.1)
 
     @pytest.mark.filterwarnings("ignore:The fit may be unsuccessful")
     @pytest.mark.filterwarnings("ignore:Model is linear in parameters")

--- a/specreduce/tracing.py
+++ b/specreduce/tracing.py
@@ -394,9 +394,13 @@ class FitTrace(Trace, _ImageParser):
             y_bins = y_bins[y_finite]
 
             # use given model to bin y-values; interpolate over all wavelengths
-            fitter = (fitting.SplineSmoothingFitter()
-                      if isinstance(self.trace_model, models.Spline1D)
-                      else fitting.LMLSQFitter())
+            if isinstance(self.trace_model, models.Spline1D):
+                fitter = fitting.SplineSmoothingFitter()
+            elif self.trace_model.linear:
+                fitter = fitting.LinearLSQFitter()
+            else:
+                fitter = fitting.LMLSQFitter()
+
             self.trace_model_fit = fitter(self.trace_model, x_bins, y_bins)
 
             trace_x = np.arange(img.shape[self._disp_axis])

--- a/specreduce/tracing.py
+++ b/specreduce/tracing.py
@@ -190,7 +190,8 @@ class FitTrace(Trace, _ImageParser):
             or `~astropy.modeling.spline.Spline1D`, optional
         The 1-D polynomial model used to fit the trace to the bins' peak
         pixels. Spline1D models are fit with Astropy's
-        'SplineSmoothingFitter', while the other models are fit with the
+        'SplineSmoothingFitter', generic linear models are fit with the
+        'LinearLSQFitter', while the other models are fit with the
         'LMLSQFitter'. [default: ``models.Polynomial1D(degree=1)``]
     peak_method : string, optional
         One of ``gaussian``, ``centroid``, or ``max``.
@@ -295,7 +296,7 @@ class FitTrace(Trace, _ImageParser):
             offset_init = models.Const1D(np.ma.median(ztot))
             profile = g1d_init + offset_init
 
-            fitter = fitting.LMLSQFitter()
+            fitter = fitting.DogBoxLSQFitter()
             popt_tot = fitter(profile, yy, ztot)
 
         # restrict fit to window (if one exists)

--- a/specreduce/tracing.py
+++ b/specreduce/tracing.py
@@ -191,7 +191,7 @@ class FitTrace(Trace, _ImageParser):
         The 1-D polynomial model used to fit the trace to the bins' peak
         pixels. Spline1D models are fit with Astropy's
         'SplineSmoothingFitter', while the other models are fit with the
-        'LevMarLSQFitter'. [default: ``models.Polynomial1D(degree=1)``]
+        'LMLSQFitter'. [default: ``models.Polynomial1D(degree=1)``]
     peak_method : string, optional
         One of ``gaussian``, ``centroid``, or ``max``.
         ``gaussian``: Fits a gaussian to the window within each bin and
@@ -295,7 +295,7 @@ class FitTrace(Trace, _ImageParser):
             offset_init = models.Const1D(np.ma.median(ztot))
             profile = g1d_init + offset_init
 
-            fitter = fitting.LevMarLSQFitter()
+            fitter = fitting.LMLSQFitter()
             popt_tot = fitter(profile, yy, ztot)
 
         # restrict fit to window (if one exists)
@@ -396,7 +396,7 @@ class FitTrace(Trace, _ImageParser):
             # use given model to bin y-values; interpolate over all wavelengths
             fitter = (fitting.SplineSmoothingFitter()
                       if isinstance(self.trace_model, models.Spline1D)
-                      else fitting.LevMarLSQFitter())
+                      else fitting.LMLSQFitter())
             self.trace_model_fit = fitter(self.trace_model, x_bins, y_bins)
 
             trace_x = np.arange(img.shape[self._disp_axis])

